pyscf/lib/libqcschema.py

```diff
@@ -0,0 +1,220 @@
+import json
+import numpy as np
+from pyscf.lib.parameters import BOHR
+
+def load_qcschema_json( file_name ):
+    # load qcschema output json file
+    data = None
+    with open(file_name,'r') as f:
+        data = json.load(f)
+    return data
+
+def load_qcschema_go_final_json( file_name ):
+    # load qcschema GO output json file 
+    # and return last 'trajectory' point's entries
+    # (this is the optimized molecule)
+    data = None
+    temp = None
+    with open(file_name,'r') as f:
+        temp = json.load(f)
+    data = temp["trajectory"][-1]
+    return data
+
+def combine_syms_coords(syms,coords,to_Angstrom = False, xyz=False):
+    # QCSchema molecules are split into symbols and coordinates,
+    #  combine them here.
+    # Optionally convert from Bohr to Angstrom.
+    # Returns: either a geometry string (default) or
+    #  optionally it can return an xyz format string.
+    if(to_Angstrom):
+        # convert Bohr to Angstrom
+        coords = coords*BOHR
+        
+    NAtoms = len(syms)
+    geo = np.reshape(coords, (NAtoms,3))
+
+    # Concatenate the symbols and coordinates along the second axis
+    combined = np.concatenate([syms[:, np.newaxis], geo], axis=1)
+
+    # Convert the combined array to a string with spaces as separators
+    output = np.array2string(combined, separator=' ', max_line_width=np.inf, threshold=np.inf)
+
+    # Remove the brackets and quotes from the output string
+    PySCF_atoms = output.replace('[', '').replace(']', '').replace("'", '')
+
+    # Return as string or return as xyz-format string (i.e. top is NAtoms,blankline)
+    if(xyz):
+        xyz = f'{NAtoms}\n\n'
+        PySCF_atoms = xyz+PySCF_atoms
+    
+    return PySCF_atoms
+
+def load_qcschema_molecule(qcschema_dict, to_Angstrom = False, xyz=False):
+    # Load QCSchema molecule.
+    # Optionally convert geometry to angstrom
+    # Returns: either a geometry string (default) or
+    #  optionally it can return an xyz format string.
+
+    syms = np.array(qcschema_dict["molecule"]["symbols"])
+    geo = np.array(qcschema_dict["molecule"]["geometry"])
+
+    # combine together the symbols and coordinates
+    PySCF_atoms = combine_syms_coords(syms,geo,to_Angstrom,xyz)
+
+    return PySCF_atoms
+
+def load_qcschema_final_molecule(qcschema_dict, to_Angstrom = False, xyz=False):
```

Comment:
Mostly duplicated to the previous function. Consider to refactor the code.

Reply:
load_qcschema_molecule now has options to load 'final_molecule', 'initial_molecule' or any molecule from a GeomOpt or Traj. It maybe isn't as clean but is in 1 func